src/pybind/mgr/dashboard/HACKING.rst

```diff
@@ -190,6 +190,29 @@ Example:
       Some <strong>helper</strong> html text
     </cd-helper>
 
+How to extract i18n messages from source code?
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+To extract the i18n messages from the templates and the TypeScript files just
+run the following command:
+
+  $ npm run i18n
+
+This will extract all marked messages from the HTML templates first and then
+add all marked strings from the TypeScript files to the translation template.
+As long as the extraction from TypeScript files is not supported from Angular
+itself, we need to use the
+[ngx-translator](https://github.com/ngx-translate/i18n-polyfill) extractor to
+parse the TypeScript files.
+
+When the command ran successfully it should have created or updated the file
+src/locale/messages.xlf. Now, you can use this file to add a new translation
+to the project or update an existing language.
+
+A new language should be placed in ``src/locale/<country-code>/messages.xlf``.
+For example the path for german would be ``src/locale/de/messages.xlf``.
```

Comment:
Does this just contain the primary language subtag (e.g. `de`)? I wonder if it would make sense to also support extended language subtags here, e.g. `en_US` vs. `en_UK`, to conform to the [IETF language tag system](https://en.wikipedia.org/wiki/IETF_language_tag)

Reply:
It supports subtags.
I have been wondering about that because of the date format. Should we allow the user to pick `en-us` and `en-uk`? Or pick only allow `en` but have another option to change the format...

Reply:
`en-US` vs `en-UK` is probably a bad example, as it's likely not going to be the first target for having different translations. But I'd assume that `en-US` is the built-in default and reference for all other languages.

If subtags are supported in general, I think we should establish using them from the beginning. The first language where this will be required is Chinese, where we plan to add locales for simplified Chinese (`zh-CN`) and traditional Chinese (`zh-TW`).

Reply:
Done.